From f092c1014d6d95cffa1bdae429dc7197c0542b1b Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Mon, 20 Mar 2017 16:37:14 -0400
Subject: [PATCH 95/97] ARM: dts: k2g-ice: add tx-internal-delay for dp83867
 clock path

K2G ICE EVM TRM recommends to add 0.25 nsec delay in the tx direction
for internal delay in the clock path to be on the safer side. Also change
the phy-mode to RGMII-ID to apply both tx and rx delays.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2g-ice.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/keystone-k2g-ice.dts b/arch/arm/boot/dts/keystone-k2g-ice.dts
index d4c858e..da4ba60 100644
--- a/arch/arm/boot/dts/keystone-k2g-ice.dts
+++ b/arch/arm/boot/dts/keystone-k2g-ice.dts
@@ -716,6 +716,7 @@
 	ethphy0: ethernet-phy@0 {
 		reg = <0>;
 		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_25_NS>;
+		ti,tx-internal-delay = <DP83867_RGMIIDCTL_250_PS>;
 		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_8_B_NIB>;
 		ti,min-output-impedance;
 	};
@@ -723,5 +724,5 @@
 
 &gbe0 {
 	phy-handle = <&ethphy0>;
-	phy-mode = "rgmii-rxid";
+	phy-mode = "rgmii-id";
 };
-- 
1.9.1

